tests/reg-test.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+expected=$(echo -e "Arguments: arg1 arg2 arg3 \nDefines: v1 v2 ")
+
+result=$(
+    cargo run -- \
+        --project ./tests/cpp \
+        --target test \
+        -D DEFINE1=v1 \
+        -D DEFINE2=v2 \
```

Comment:
Should be -DDEFINE1=v1

Reply:
Done. There are three ways:
* `-DDEFINE=v1`
* `-D DEFINE=v1` (not forbidden, it's a bit more complicated than one would think)
* `--cmake-define DEFINE=v1`